src/Jobs/ChannelMonitorJob.cs

```diff
@@ -0,0 +1,152 @@
+/*
+ * NodeGuard
+ * Copyright (C) 2023  Elenpay
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+using FundsManager.Data;
+using FundsManager.Data.Models;
+using FundsManager.Data.Repositories.Interfaces;
+using FundsManager.Services;
+using Google.Protobuf;
+using Grpc.Core;
+using Lnrpc;
+using Microsoft.EntityFrameworkCore;
+using Quartz;
+using Channel = Lnrpc.Channel;
+
+namespace FundsManager.Jobs;
+
+/// <summary>
+/// Job for update the status of the channels
+/// </summary>
+/// <returns></returns>
+[DisallowConcurrentExecution]
+public class ChannelMonitorJob : IJob
+{
+    private readonly ILogger<ChannelMonitorJob> _logger;
+    private readonly IDbContextFactory<ApplicationDbContext> _dbContextFactory;
+    private readonly INodeRepository _nodeRepository;
+    private readonly ILightningService _lightningService;
+
+    public ChannelMonitorJob(ILogger<ChannelMonitorJob> logger, IDbContextFactory<ApplicationDbContext> dbContextFactory, INodeRepository nodeRepository, ILightningService lightningService)
+    {
+        _logger = logger;
+        _dbContextFactory = dbContextFactory;
+        _nodeRepository = nodeRepository;
+        _lightningService = lightningService;
+    }
+
+    public async Task Execute(IJobExecutionContext context)
+    {
+        var data = context.JobDetail.JobDataMap;
+        var nodeId = data.GetInt("nodeId");
+        _logger.LogInformation("Starting {JobName} for node {nodeId}... ", nameof(ChannelMonitorJob), nodeId);
+        try
+        {
+            var node1 = await _nodeRepository.GetById(nodeId);
+
+            if (node1 == null)
+            {
+                _logger.LogInformation("The node {NodeId} was set up for monitoring but the node doesn't exist anymore", nodeId);
+                return;
+            }
+
+            var client = LightningService.CreateLightningClient(node1.Endpoint);
```

Comment:
Hmm.. creating a gRPC client all the time can lead to resources exhaustion, we should add them via DI

Reply:
Will investigate

Reply:
I created a service for reusing clients in jobs.
Other parts of the application (like the lightning service) create the lightning client inside static functions, so we will have to refactor this in order to reuse the client there.

Reply:
Why not this? https://learn.microsoft.com/en-us/aspnet/core/grpc/clientfactory?view=aspnetcore-7.0